code/game/turfs/flooring/flooring.dm

```diff
@@ -35,52 +39,217 @@ var/list/flooring_types
 	var/build_time = 0  // BYOND ticks.
 
 	var/descriptor = "tiles"
-	var/flags
+	var/flags = TURF_CAN_BURN | TURF_CAN_BREAK
 	var/can_paint
 
+	var/is_plating = FALSE
+
+	//Plating types, can be overridden
+	var/plating_type = /decl/flooring/reinforced/plating
+
+	//Resistance is subtracted from all incoming damage
+	var/resistance = 3
+
+	//Damage the floor can take before being destroyed
+	var/health = 50
+
+	var/removal_time = WORKTIME_FAST * 0.75
+
+
+//Flooring Procs
+/decl/flooring/proc/get_plating_type(var/turf/location)
+	return plating_type
+
+
+//Used to check if we can build the specified type of floor ontop of this one
+/decl/flooring/proc/can_build_floor(var/decl/flooring/newfloor)
+	return FALSE
+
+
+//Used when someone attacks the floor
+/decl/flooring/proc/attackby(var/obj/item/I, var/mob/user, var/turf/T)
+	return FALSE
+
+
+/decl/flooring/proc/Entered(mob/living/M as mob)
+	return
+
+
 /decl/flooring/grass
 	name = "grass"
 	desc = "Do they smoke grass out in space, Bowie? Or do they smoke AstroTurf?"
 	icon = 'icons/turf/flooring/grass.dmi'
 	icon_base = "grass"
 	has_base_range = 3
 	damage_temperature = T0C+80
-	flags = TURF_REMOVE_SHOVEL
+	flags = TURF_REMOVE_SHOVEL | SMOOTH_ONLY_WITH_ITSELF | TURF_HAS_EDGES | TURF_EDGES_EXTERNAL
 	build_type = /obj/item/stack/tile/grass
+	plating_type = /decl/flooring/dirt
+	footstep_sound = "grass"
+
+/decl/flooring/dirt
+	name = "dirt"
+	desc = "Do they smoke grass out in space, Bowie? Or do they smoke AstroTurf?"
+	icon = 'icons/turf/flooring/dirt.dmi'
+	icon_base = "dirt"
+	flags = TURF_REMOVE_SHOVEL | SMOOTH_ONLY_WITH_ITSELF
+	build_type = null //Todo: add bags of fertilised soil or something to create dirt floors
+	footstep_sound = "gravel"
 
 /decl/flooring/asteroid
 	name = "coarse sand"
 	desc = "Gritty and unpleasant."
 	icon = 'icons/turf/flooring/asteroid.dmi'
 	icon_base = "asteroid"
-	flags = TURF_HAS_EDGES | TURF_REMOVE_SHOVEL
+	flags = TURF_HAS_EDGES | TURF_REMOVE_SHOVEL | TURF_CAN_BURN | TURF_CAN_BREAK
```

Comment:
Burn and break flags are applied a lot more extensively. They simply mean that the turf can recieve those overlays, not necessarily that it's flammable